### PR TITLE
Bump up to jakartified LRA 2.0

### PR DESCRIPTION
**Pull request by danielkec**
Bump up LRA to 2.0-RC1 release

Signed-off-by: Daniel Kec <daniel.kec@oracle.com>

### CODE REVIEW

File: config/config/src/main/java/io/helidon/config/ConfigHelper.java [hunk 1]
```diff
@@ -43,7 +43,7 @@ private ConfigHelper() {
                 .filter(e -> e.getValue() instanceof ValueNodeImpl)
                 .collect(Collectors.toMap(
                         e -> e.getKey().toString(),
-                        e -> Config.Key.escapeName(((ValueNodeImpl) e.getValue()).get())
+                        e -> ((ValueNodeImpl) e.getValue()).get()
```
Comment (tomas-langer) [inline config/config/src/main/java/io/helidon/config/ConfigHelper.java:46]: Why is this change done?

File: microprofile/tests/tck/tck-lra/src/test/java/io/helidon/microprofile/lra/tck/CoordinatorDeployer.java [hunk 2]
```diff
@@ -52,18 +54,21 @@ public void beforeStart(@Observes BeforeStart event, Container container) throws
         HelidonContainerConfiguration containerConfig = helidonContainer.getContainerConfig();
 
         containerConfig.addConfigBuilderConsumer(configBuilder -> {
-            configBuilder.withSources(MpConfigSources.create(CoordinatorService.class.getResource("/application.yaml")),
+            var is = CoordinatorService.class.getResourceAsStream("/application.yaml");
+            configBuilder.withSources(MpConfigSources.create(ConfigSources.create(is, MediaType.APPLICATION_X_YAML.toString())),
```
Comment (tomas-langer) [inline microprofile/tests/tck/tck-lra/src/test/java/io/helidon/microprofile/lra/tck/CoordinatorDeployer.java:58]: why do you use X_YAML and not YAML?

**Review by tomas-langer — CHANGES_REQUESTED** [review]
The commit for escaping config values is not ok, we need to fix that first.

File: microprofile/tests/tck/tck-lra/src/test/java/io/helidon/microprofile/lra/tck/CoordinatorDeployer.java (hunk 2 above)
Comment (danielkec) [reply microprofile/tests/tck/tck-lra/src/test/java/io/helidon/microprofile/lra/tck/CoordinatorDeployer.java:58]: I need correct config parser to kick in https://github.com/oracle/helidon/blob/master/config/yaml/src/main/java/io/helidon/config/yaml/YamlConfigParser.java#L55

File: config/config/src/main/java/io/helidon/config/ConfigHelper.java (hunk 1 above)
Comment (danielkec) [reply config/config/src/main/java/io/helidon/config/ConfigHelper.java:46]: There was this issue with escaped dot in config value:
```java
Map<String, String> values = Map.of("key.some", "jdbc:h2:file:./target/lra-coordinator");
        org.eclipse.microprofile.config.spi.ConfigSource mpSource = MpConfigSources.create(ConfigSources.create(values).build());
        assertThat(mpSource.getValue("key.some"), is("jdbc:h2:file:./target/lra-coordinator"));
```
```
Expected :jdbc:h2:file:./target/lra-coordinator
Actual   :jdbc:h2:file:~1/target/lra-coordinator
```
Escape is moved here:
https://github.com/oracle/helidon/pull/3769/files#diff-9d50a6dc46f6009326acc97b03ec1f183f0018408926cfb98eb236ebb8c85951R184

**Comment by danielkec** [comment]
Escaping fixed

File: config/config/src/test/java/io/helidon/config/KeyTokenResolvingTest.java [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2017, 2021 Oracle and/or its affiliates.
+ * Copyright (c) 2017, 2022 Oracle and/or its affiliates.
```
Comment (spericas) [inline config/config/src/test/java/io/helidon/config/KeyTokenResolvingTest.java:2]: Only copyright updated on this file

Comment (danielkec) [reply config/config/src/test/java/io/helidon/config/KeyTokenResolvingTest.java]: Sorry for the force-push, had to do the squash

**Review by spericas — APPROVED** [review]

**Review by tomas-langer — APPROVED** [review]
LGTM

**Pull request merged**